pkg/templates/generic_lookup_func.go

```diff
@@ -77,8 +77,9 @@ func (t *TemplateResolver) lookup(
 		getObj, lookupErr := dclient.Get(context.TODO(), rsrcname, metav1.GetOptions{})
 		if lookupErr == nil {
 			result = getObj.UnstructuredContent()
-			t.addToReferencedObjects(getObj.GetAPIVersion(), getObj.GetKind(), ns, getObj.GetName())
 		}
+
+		t.addToReferencedObjects(apiversion, kind, ns, rsrcname)
```

Comment:
Same comment as above (since I think this also misses the list lookup below).

Reply:
Same comment as above except I forgot to guard against unexpected errors.